src/mono/msbuild/apple/data/ProxyProjectForAOTOnHelix.proj

```diff
@@ -12,6 +12,8 @@
     <!-- Forced by ILLink targets -->
     <SelfContained>true</SelfContained>
     <PublishDir>$(OriginalPublishDir)</PublishDir>
+    <!-- Prevent getting DynamicCodeSupport=true default from ILLink targets that are imported with the Sdk.targets -->
+    <DynamicCodeSupport>false</DynamicCodeSupport>
```

Comment:
Nice catch! These are difficult to detect when something goes wrong at runtime.

Reply:
I was under the impression we'd only do AOT on helix, not actual trimming. @steveisok how does this work?

Reply:
The new default should affect many more cases than just AOT on helix.

Reply:
> I was under the impression we'd only do AOT on helix, not actual trimming. @steveisok how does this work?

If you want trimming + AOT on helix, both have to occur in the same place. Otherwise, you have to send individual copies of the trimmed shared framework for each test from the build machine.

Reply:
That might not be as painful as it seemed initially.

Reply:
@akoeplinger Since we perform trimming on Helix, we can move this to the `Directory.Build.props` file. However, it will not have effect on other components.

Reply:
I've added a default switch to disable trimming on build machines as well. I haven't found a common location where these switches are set in Mono. Since we don't ship Apple mobile support from the runtime repository, the switch is set in the SDK/Xamarin.

A potential place to set this switch could be the AOT compiler props, but they are invoked after the trimming.

/cc: @sbomer

Reply:
The ILLink feature switches are moved to `ConfigureTrimming` target.